### DISCUSSION TITLE
I dont get how to use this findUnique

### DISCUSSION THREAD

**Discussion by BJladika** [prisma/prisma#7533] [Q&A] [answered]
So... I have a types Student (aka peoples) and Nationality (which serves as a directory like returning if id_Nationality = 1 then German, and etc..)
```prisma
model people_List {
  id_People           Int              @id
  Surname             String           @db.VarChar(30)
  Name                String           @db.VarChar(30)
  Patronymic          String?          @db.VarChar(30)
  id_Nationality      Int?
  spr_Nationality     spr_Nationality? @relation(fields: [id_Nationality], references: [id_Nationality])
}
```
```
type Student {
  id_People: ID!
  Name: String
  Surname: String
  Patronymic: String
  FullName: String
  Nationality: Nationality
  id_Nationality: ID
}
```
```prisma
model spr_Nationality {
  id_Nationality Int           @id
  Nationality    String        @db.VarChar(100)
  people_List    people_List[]
}
```
```
type Nationality{
  id_Nationality: ID!
  Nationality: String
}
```
I tried to make a resolver (based on examples like Post having 1 Author)
```ts
    Nationality: (parent, _args, ctx) => {
      return ctx.prisma.spr_Nationality
        .findUnique({
          where: { id_Nationality: parent.id_Nationality },
        })
        .Nationality()
    },
```
But I'm receiving this Message "ctx.prisma.spr_Nationality.findUnique(...).Nationality is not a function""
Where I did go wrong?

UPD: Tried to do also
```ts
      return ctx.prisma.spr_Nationality
        .findUnique({
          where: { id_Nationality: parent.id_Nationality },
        })
        .spr_Nationality()
```
but still error

**Comment by ryands17** [comment, marked as answer]
@BJladika 👋 

Your case is not similar to a Post having one author. If you just want to return Nationality, then simply this is only required:

```ts
return  ctx.prisma.spr_Nationality
    .findUnique({
      where: { id_Nationality: parent.id_Nationality },
    })
```

**Reply by BJladika** [reply]
I tried like that, and now receiving other error (yes, rows of Student can have possibly have null id_Nationality)
```"message": "\nInvalid `prisma.spr_Nationality.findUnique()` invocation:\n\n{\n  where: {\n    id_Nationality: null\n                    ~~~~\n  }\n}\n\nArgument id_Nationality for where.id_Nationality must not be null. Please use undefined instead.\n\n",```
I also tried to write like somewhere was written, but then I receive other error

```
return ctx.prisma.spr_Nationality
        .findUnique({
          where: { id_Nationality: parent.id_Nationality != null ? parent.id_Nationality : undefined},
        })
```
```
"message": "\nInvalid `prisma.spr_Nationality.findUnique()` invocation:\n\n{\n  where: {\n?   id_Nationality?: Int\n  }\n}\n\nArgument where of type spr_NationalityWhereUniqueInput needs at least one argument. Available args are listed in green.\n\nNote: Lines with ? are optional.\n"
```

**Reply by ryands17** [reply]
You cannot pass `null` there. So you need to use:

```ts
if (parent.id_Nationality)
return  ctx.prisma.spr_Nationality
    .findUnique({
      where: { id_Nationality: parent.id_Nationality },
})

return null
```

**Reply by BJladika** [reply]
Okay... seems it works, but now I've encountered other error with other field (when tried to add new type to gql), should I start new topic or continue there?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
